app/models/topic_hot_score.rb

```diff
@@ -5,6 +5,22 @@ class TopicHotScore < ActiveRecord::Base
 
   DEFAULT_BATCH_SIZE = 1000
 
+  def self.hot_topic_ids_cache
+    @hot_topic_ids_cache ||= DistributedCache.new("hot_topic_ids_cache")
+  end
+
+  def self.hottest_topic_ids
+    hot_topic_ids_cache["ids"].to_a
+  end
+
+  def self.recreate_hottest_topic_ids_cache
+    hot_topic_ids_cache.clear
```

Comment:
Is there any way to avoid clearing before we do the `get_set` so there is no chance of a request coming through after we clear, before we set?

Reply:
Right so every 10 minutes there is a very slight window where `hot_topic_ids_cache` may be empty, right?

Reply:
Technically, yes. It's not an atomic operation. However, since we'll use this to display a badge in the UI, I think it's fine if the cache is empty during that window.

Reply:
Cool cool, I think it's fine. Just if there was another way of doing it with a smaller/no gap then figured I mention it.

Reply:
I'll see if there's a better way of doing it. Tried using `defer_get_set`, which should overwrite the entry, but I was getting old data during testing.